assignments/assignment04.rb

```diff
@@ -40,26 +41,85 @@ def fib(n)
 
 class Queue
   def initialize
-    # your implementation here
+    @wholepile = nil
+    @bottom = nil
   end
-  def enqueue(item)
-    # your implementation here
+
+  class Piece
+    attr :chunk, :therest
+    def initialize(chunk, therest)
+      @chunk = chunk
+      @therest = therest
+    end
+    def chunk=(new_chunk)
+      @chunk = new_chunk
+    end
+    def therest=(new_therest)
+      @therest = new_therest
+    end
+  end
+
+  def enqueue(chunk)
+    @wholepile = Piece.new chunk, @wholepile
+    self
   end
+
   def dequeue
-    # your implementation here
+    def qDigger(leveldown)
```

Comment:
ruby methods have snake_case names, never camelCase. So this should be called `q_digger`